### PR TITLE
Hamid

### PR DESCRIPTION
**Pull request by HamidNiakan**
task completed

### CODE REVIEW

**Review by yeganemehr — CHANGES_REQUESTED** [review]
Hi Hamid
Thank you for participating in the challenge.
You must not change the tests, Please revert your changes.
Also put your code on correct directory like it's described on Readme.md

**Comment by HamidNiakan** [comment]
چشم حتما انجام میدم

File: challenges/3/YOUR-CODE-GOES-HERE/hamidrezaNiakan/app/Http/Controllers/TravelController.php [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+
+class TravelController extends Controller {
+	public function view ( Travel $travel , Request $request ) {
+		$user = $request->user();
+		if ( $user->can('view' , $travel) ) {
+			return [
+				'travel' => [
+					'id' => $travel->id ,
```
Comment (yeganemehr) [inline challenges/3/YOUR-CODE-GOES-HERE/hamidrezaNiakan/app/Http/Controllers/TravelController.php:27]: What is point of this controller if you give it travel's id and retrieve just id of the travel?
One of soft skills you need in every job you'll need is that you do tasks based on context.

File: challenges/3/YOUR-CODE-GOES-HERE/hamidrezaNiakan/app/Http/Controllers/TravelSpotController.php [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+			if ( $travel->status->value == TravelStatus::CANCELLED->value ) {
+				throw new InvalidTravelStatusForThisActionException();
+			}
+			$origin_Spot = $travel->getOriginSpot();
+			$origin_Spot->arrived_at = Carbon::now();
+			$origin_Spot->save();
+			
+			return response()->json(TravelResource::make($travel));
+		}
+		DB::commit();
```
Comment (yeganemehr) [inline challenges/3/YOUR-CODE-GOES-HERE/hamidrezaNiakan/app/Http/Controllers/TravelSpotController.php:37]: You commit the transaction after you return. Tests are passed but not going to work on production.
It's a state of art!

File: challenges/3/YOUR-CODE-GOES-HERE/hamidrezaNiakan/app/Http/Controllers/TravelSpotController.php [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	}
+	
+	public function store ( Travel $travel , Request $request ) {
+		$this->validate($request , [
```
Comment (yeganemehr) [inline challenges/3/YOUR-CODE-GOES-HERE/hamidrezaNiakan/app/Http/Controllers/TravelSpotController.php:41]: We already provide you `TravelSpotStoreRequest` to help you with validation.
This apply for all of other insert/update controllers too.

File: challenges/3/YOUR-CODE-GOES-HERE/hamidrezaNiakan/app/Http/Controllers/TravelController.php [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+				] ,
+			];
+		}
+	}
+	
+	public function store ( TravelStoreRequest $request ) {
```
Comment (yeganemehr) [inline challenges/3/YOUR-CODE-GOES-HERE/hamidrezaNiakan/app/Http/Controllers/TravelController.php:33]: Authorization and policies are forgotten?

**Review by yeganemehr — APPROVED** [review]
Thank you for following up the challenge.
I'll merge it.
HR call soon.

**Comment by HamidNiakan** [comment]
سلام وقت بخیر این ها رو برطرف خواهم کردم

**Comment by HamidNiakan** [comment]
سلام وقت بخیر
مواردی که در merge request گفته بودید رو برطرف کردم .
می خواستم ببینم نتیجه کی مشخص میشه؟

**Pull request merged**